torch/onnx/_internal/exporter.py

```
@@ -382,18 +442,73 @@ def save(
+                destination is "/path/model.onnx", the initializers will be saved in "/path/model_initializers/" folder.
+            model_state_dict: The state_dict of the PyTorch model with all weights on it (e.g. `model.state_dict()`).
+                Required when ``enable_fake_mode`` is used but real initializers are needed on the ONNX graph.
+                It can be either a string with the path to a checkpoint or a dictionary with the actual model state.
+            serializer: The serializer to use. If not specified, the model will be serialized as Protobuf.
+        """
 
         if serializer is None:
             serializer = ProtobufExportOutputSerializer()
-        if isinstance(destination, str):
-            with open(destination, "wb") as f:
-                serializer.serialize(self, f)
+
+        # Add initializers when symbolic tracing is enabled
+        _model_state_dict_files = None
+        if model_state_dict is not None:
+            if isinstance(model_state_dict, dict):
+                model_state_dict_file = io.BytesIO()
+                torch.save(model_state_dict, model_state_dict_file)
+                model_state_dict_file.seek(0)
+            else:
+                isinstance(
+                    model_state_dict, str
+                ), "model_state_dict must be a path to the model's state_dict or the actual state_dict"
+                model_state_dict_file = model_state_dict
+
+            ctx = patcher.ONNXTorchPatcher()
+            with ctx:
+                _ = torch.load(model_state_dict_file)
+                _model_state_dict_files = ctx.paths
+
+            # Reset the buffer to the beginning before reading it again
+            if isinstance(model_state_dict, dict):
+                for file_obj in ctx.paths:
+                    file_obj.seek(0)  # type: ignore[union-attr]
+
+        if _model_state_dict_files is not None:
+            if not isinstance(destination, str):
+                raise RuntimeError(
```

this check can be moved up to fail early